src/include/file.h

```diff
@@ -0,0 +1,123 @@
+#ifndef FILE_H
+#define FILE_H
+
+#include <dirent.h>
+#include <stdio.h>
+#include <signal.h>
+#include <string.h>
+#include <stdlib.h>
+#include <libgen.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+void retrieve_downloadable_filenames(char **files)
+{
+    int filesLength = 10;
+    int fileCount = 0;
+
+    DIR *dir;
+    struct dirent *ent;
+    if ((dir = opendir("../src/server/files")) != NULL)
```

Comment:
Shouldn't be hardcoded
```suggestion
    if ((dir = opendir(directory)) != NULL)
```